src/sync.rs

```diff
@@ -1,15 +1,81 @@
+#[cfg(all(not(feature = "std"), loom))]
+compile_error!("loom requires the standard library");
+
+//loom support
+
 #[cfg(loom)]
 pub(crate) mod atomic {
-    pub(crate) use loom::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use loom::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
 }
-
 #[cfg(loom)]
 pub(crate) use loom::thread::yield_now;
 
-#[cfg(not(loom))]
+//std imports
+
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) mod atomic {
-    pub(crate) use std::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use std::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
 }
-
-#[cfg(not(loom))]
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) use std::thread::yield_now;
+
+//both
+
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod marker {
+    pub(crate) use std::marker::PhantomData;
+}
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod ptr {
+    pub(crate) use std::ptr::{null, null_mut, NonNull, drop_in_place};
+}
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod mem {
+    pub(crate) use std::mem::transmute;
+}
+#[cfg(any(loom, feature = "std"))]
+pub(crate) mod boxed {
+    pub(crate) use std::boxed::Box;
```

Comment:
Same here: we should just use `alloc::` (instead of `std::`) everywhere directly. You'll need to add a `extern crate alloc` to `src/lib.rs`, but that's fine.

Reply:
Good to know. Thank you!